jdaviz/configs/default/plugins/virtual_observatory/vo_plugin.py

```diff
@@ -0,0 +1,419 @@
+
+        # Obtain center point of the current image and convert into sky coordinates
+        x_center = (viewer.state.x_min + viewer.state.x_max) * 0.5
+        y_center = (viewer.state.y_min + viewer.state.y_max) * 0.5
+        skycoord_center = viewer.state.reference_data.coords.pixel_to_world(
+            x_center, y_center
+        )
+
+        # Extract SkyCoord values as strings for plugin display
+        ra_deg = skycoord_center.ra.deg
+        dec_deg = skycoord_center.dec.deg
+        frame = skycoord_center.frame.name.lower()
+
+        # Show center value in plugin
+        self.source = f"{ra_deg} {dec_deg}"
+        self.coordframe_selected = frame
+
+        self.viewer_centered = True
+
+    @observe("waveband_selected", "change")
+    def vue_query_registry_resources(self, _=None):
+        """
+        Query Virtual Observatory registry for all SIA services
+        that serve data in that waveband around the source.
+        Then update the dropdown accordingly.
+        """
+        # If waveband was changed to nothing, immediately quit
+        # Don't throw an error due to trigger by plugin init
+        if not self.waveband_selected:
+            return
+
+        # Can't filter by coverage if we don't have a source to filter on
+        if self.resource_filter_coverage and not self.source:
+            error_msg = (
+                "Source is required for registry querying when coverage filtering is enabled. "
+                + (
+                    "Please enter your coordinates above "
+                    if self.viewer == "Manual"
+                    else f"Load data into viewer {self.viewer} first before querying "
+                )
+                + "or disable coverage filtering."
+            )
+            self.hub.broadcast(SnackbarMessage(error_msg, sender=self, color="error"))
+            raise ValueError(error_msg)
+
+        # Clear existing resources list
+        self.resources = []
+        self.resource_selected = None
+        self.resources_loading = True  # Start loading bar
+
+        try:
+            registry_args = [
+                registry.Servicetype("sia"),
+                registry.Waveband(self.waveband_selected),
+            ]
+            # If coverage filtering is enabled, lookup current
+            # source coordinate and add a Spatial search constraint
+            if self.resource_filter_coverage:
+                try:
+                    # First parse user-provided source as direct coordinates
+                    coord = SkyCoord(
+                        self.source, unit=u.deg, frame=self.coordframe_selected
+                    )
+                except Exception:
+                    try:
+                        # If that didn't work, try parsing it as an object name
+                        coord = SkyCoord.from_name(
+                            self.source, frame=self.coordframe_selected
+                        )
+                    except Exception:
```

Comment:
can either of these not be bare-exceptions (just to avoid triggering on small bugs/typos)?  Or are there too many things that could raise exceptions to catch explicitly?

Reply:
The main reason for this behavior is to create a snackbar message indicating the error, since exceptions at the plugin level that are initiated from the UI don't actually appear in the notebook. That's why you can see Me create a snackbar message then immediately reraise another error. I tried to add a level of abstraction by providing a new error type/message to help guide the user to what's going on; otherwise, I would have just reraised the exact same exception.

It's not really to catch a specific error, it's designed explicitly to inject an additional snackbar message to whatever existing exceptions bubble up.